lambdas/service/app.py

```diff
@@ -1611,3 +1616,35 @@ def get_data_object(file_uuid):
         return Response({'data_object': data_obj}, status_code=200)
     else:
         return Response({'msg': "Data object not found."}, status_code=404)
+
+
+def gs_url(file_uuid, version):
+    url = config.dss_endpoint + '/files/' + urllib.parse.quote(file_uuid, safe='')
+    params = dict({'file_version': version} if version else {},
+                  directurl=True,
+                  replica='gcp')
+    while True:
+        if app.lambda_context.get_remaining_time_in_millis() / 1000 > 3:
+            dss_response = requests.get(url, params=params, allow_redirects=False)
+            if dss_response.status_code == 302:
+                url = dss_response.next.url
+                assert url.startswith('gs')
+                return url
+            elif dss_response.status_code == 301:
+                url = dss_response.next.url
+                remaining_lambda_seconds = app.lambda_context.get_remaining_time_in_millis() / 1000
+                server_side_sleep = min(
+                    10, max(remaining_lambda_seconds - config.api_gateway_timeout_padding - 3, 0))
```

Comment:
The default minimum sleep of 10 seconds before retrying should be 1 second instead.
Given this endpoint primarily exists for the HCA/Terra handoff MVP, and this is all on Google, and the DSS checkout time for Google is typically very fast (<= 1 second) this endpoint should retry the request very quickly (sleep time of 1 second).
We had discussed this with the Broad and verbally agreed on this as the plan, yet it had not yet been incorporated into our documentation. Sorry about that, I am working on incorporating that now.